src/factory/providers/zksync-era/increment/index.ts

```diff
@@ -23,27 +40,39 @@ async function tvl(params: ITvlParams): Promise<Partial<ITvlReturn>> {
     web3,
   );
 
-  const collaterals = await util.executeMultiCallsOfTarget(
-    VAULT_ADDRESS,
-    VAULT_ABI,
-    'getWhitelistedCollateral',
-    Array.from({ length: numCollaterals }, (v, i) => [i]),
-    block,
-    chain,
-    web3,
-  );
+  if (collaterals.length < numCollaterals) {
+    const newCollaterals = await util.executeMultiCallsOfTarget(
+      VAULT_ADDRESS,
+      VAULT_ABI,
+      'getWhiteListedCollateral',
+      Array.from(Array(numCollaterals - collaterals.length), (_, i) => [
+        i + collaterals.length,
+      ]),
+      block,
+      chain,
+      web3,
+    );
+
+    collaterals = [...collaterals, ...newCollaterals];
+    await basicUtil.saveIntoCache(
+      collaterals,
+      'cache/collaterals.json',
+      chain,
+      provider,
+    );
+  }
 
   const balanceResults = await util.getTokenBalances(
     VAULT_ADDRESS,
-    collaterals.map((collateral) => collateral.asset),
```

Comment:
I was getting the following error before changing `collateral.asset` to `collateral[0]`:
`{"statusCode":500,"message":"Cannot read properties of undefined (reading 'balance')"}`

Reply:
Ah ok, yeah that makes sense. The Collateral struct was just not decoded properly, your solution works best